modules/shadertools/src/lib/inject-shader.js

```diff
@@ -18,44 +19,52 @@ export default function injectShader(source, type, inject, injectStandardStubs)
   const isVertex = type === VERTEX_SHADER;
 
   for (const key in inject) {
-    const fragment = inject[key];
+    const fragmentData = inject[key];
+    fragmentData.sort((a, b) => a.order - b.order);
+    fragments.length = fragmentData.length;
+    for (let i = 0, len = fragmentData.length; i < len; ++i) {
+      fragments[i] = fragmentData[i].injection;
+    }
+    const fragmentString = `${fragments.join('\n')}\n`;
     switch (key) {
       // declarations are injected before the main function
       case 'vs:#decl':
         if (isVertex) {
-          source = source.replace(REGEX_START_OF_MAIN, match => `${fragment}\n${match}`);
+          source = source.replace(REGEX_START_OF_MAIN, match => `${fragmentString}\n${match}`);
         }
         break;
       // main code is injected at the end of main function
       case 'vs:#main-start':
         if (isVertex) {
-          source = source.replace(REGEX_START_OF_MAIN, match => match + fragment);
+          source = source.replace(REGEX_START_OF_MAIN, match => match + fragmentString);
         }
         break;
       case 'vs:#main-end':
         if (isVertex) {
-          source = source.replace(REGEX_END_OF_MAIN, match => fragment + match);
+          source = source.replace(REGEX_END_OF_MAIN, match => fragmentString + match);
         }
         break;
       case 'fs:#decl':
         if (!isVertex) {
-          source = source.replace(REGEX_START_OF_MAIN, match => `${fragment}\n${match}`);
+          source = source.replace(REGEX_START_OF_MAIN, match => `${fragmentString}\n${match}`);
         }
         break;
       case 'fs:#main-start':
         if (!isVertex) {
-          source = source.replace(REGEX_START_OF_MAIN, match => match + fragment);
+          source = source.replace(REGEX_START_OF_MAIN, match => match + fragmentString);
         }
         break;
       case 'fs:#main-end':
         if (!isVertex) {
-          source = source.replace(REGEX_END_OF_MAIN, match => fragment + match);
+          source = source.replace(REGEX_END_OF_MAIN, match => fragmentString + match);
         }
         break;
 
       default:
+        // TODO(Tarek): I think this usage should be deprecated.
```

Comment:
Can we add a test case for this? Several deck.gl layers are still depending on this feature.

Reply:
Done.